vaadin-grid-flow-parent/vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/MockUI.java

```diff
@@ -0,0 +1,91 @@
+package com.vaadin.flow.component.grid;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.VaadinSession;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class MockUI extends UI {
```

Comment:
The v23 version uses a public class from a test in the Flow framework itself (`com.vaadin.flow.data.provider.DataCommunicatorTest`), which does not exist in v14. Copied the class over, which seems to work fine.